diff -Naur chicken/Makefile.linux chicken.new/Makefile.linux
--- chicken/Makefile.linux	2009-08-01 02:06:10.000000000 +0400
+++ chicken.new/Makefile.linux	2009-08-13 20:46:36.000000000 +0400
@@ -33,15 +33,15 @@
 
 # options
 
-C_COMPILER_OPTIONS = -fno-strict-aliasing -DHAVE_CHICKEN_CONFIG_H
+C_COMPILER_OPTIONS = -fexceptions -fno-strict-aliasing -DHAVE_CHICKEN_CONFIG_H
 ifdef DEBUGBUILD
 C_COMPILER_OPTIMIZATION_OPTIONS ?= -g -Wall -Wno-unused
 else
 C_COMPILER_OPTIMIZATION_OPTIONS ?= -Os -fomit-frame-pointer
 endif
 LINKER_LINK_SHARED_LIBRARY_OPTIONS = -shared
-LINKER_LINK_SHARED_DLOADABLE_OPTIONS = -L. -shared -Wl,-R$(RUNTIME_LINKER_PATH)
-LINKER_LINK_SHARED_PROGRAM_OPTIONS = -Wl,-R$(RUNTIME_LINKER_PATH)
+LINKER_LINK_SHARED_DLOADABLE_OPTIONS = -L. -shared
+LINKER_LINK_SHARED_PROGRAM_OPTIONS =
 LIBRARIES = -lm -ldl
 NEEDS_RELINKING = yes
 
